src/indexes.rs

```diff
@@ -72,14 +52,41 @@ impl JsonIndex {
 pub struct Index {
     pub(crate) uid: Rc<String>,
     pub(crate) client: Client,
+    pub(crate) primary_key: Option<String>,
+    pub created_at: Option<OffsetDateTime>,
+    pub updated_at: Option<OffsetDateTime>,
 }
 
 impl Index {
+    /// Internal Function to create an [Index] from `serde_json::Value` and [Client]
+    pub(crate) fn from_value(v: serde_json::Value, client: Client) -> Result<Index, Error> {
+        #[derive(Deserialize, Debug)]
+        #[allow(non_snake_case)]
+        struct PrivIndex {
```

Comment:
What do you think to use `InternalIndex`, `IndexFromSerde` or something like that?

Reply:
Regarding the name of the struct? `IndexFromSerde` makes sense actually.

Reply:
Yes, about the name of the struct :)

Reply:
Done!